### ISSUE TITLE
Integrating graphql-query-complexity

### ISSUE THREAD

**Issue by aldis-ameriks** [mcollina/fastify-gql#188]
Hello,
I'm looking at the option of switching from apollo fastify to fastify-gql. I'm using [graphql-query-complexity](https://github.com/slicknode/graphql-query-complexity) which I'm able to install via apollo plugins like so

```
  const apolloServer = new ApolloServer({
    schema,
    plugins: [
      {
        requestDidStart: (_requestContext) => ({
          didResolveOperation({ request, document }) {
            const complexity = getComplexity({
              schema,
              operationName: request.operationName,
              query: document,
              variables: request.variables,
              estimators: [fieldExtensionsEstimator(), simpleEstimator({ defaultComplexity: 1 })],
            });
            if (complexity >= 30) {
              throw new Error(
                `Sorry, too complicated query! ${complexity} is over 30 that is the max allowed complexity.`
              );
            }
          },
        }),
      },
    ],
  });
```

What would be an equivalent version with fastify-gql?

Thanks!

**Comment by mcollina** [comment]
In order to integrate this we’ll need to add the equivalent of validationRules. Would you like to send a PR for that?

**Comment by aldis-ameriks** [comment]
@mcollina I see. Unfortunately, I'm a bit swamped right now. If time permits, I may engage at a later time. Thanks!

**Comment by aldis-ameriks** [comment]
This is is now resolved via #189 and #193

Here's an example how to integrate graphql-query-complexity.
```
import GQL from 'fastify-gql';
import queryComplexity, { fieldExtensionsEstimator, simpleEstimator } from 'graphql-query-complexity';

app.register(GQL, {
    schema,
    validationRules: ({ variables, operationName }) => [
      queryComplexity({
        maximumComplexity: 100,
        variables,
        operationName,
        estimators: [fieldExtensionsEstimator(), simpleEstimator({ defaultComplexity: 1 })],
        onComplete: (complexity: number) => {
          console.log('Query Complexity:', complexity);
        },
      }),
    ],
  });
```
